jax_plugins/rocm/__init__.py

```diff
@@ -0,0 +1,92 @@
+# Copyright 2023 The JAX Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import functools
+import importlib
+import logging
+import os
+import pathlib
+import platform
+import sys
+
+from jax._src.lib import xla_client
+import jax._src.xla_bridge as xb
+
+# rocm_plugin_extension locates inside jaxlib. `jaxlib` is for testing without
+# preinstalled jax rocm plugin packages.
+for pkg_name in ['jax_rocm60_plugin', 'jaxlib']:
+  try:
+    rocm_plugin_extension = importlib.import_module(
+        f'{pkg_name}.rocm_plugin_extension'
+    )
+  except ImportError:
+    rocm_plugin_extension = None
+  else:
+    break
+
+logger = logging.getLogger(__name__)
+
+
+def _get_library_path():
+  installed_path = (
+      pathlib.Path(__file__).resolve().parent / 'xla_rocm_plugin.so'
+  )
+  if installed_path.exists():
+    return installed_path
+
+  local_path = os.path.join(
```

Comment:
Consider re-using the `Path` object from L43 here and below. That will allow you to express all file operations via `pathlib` without mixing in `os.path`.